src/Tizen.Network.Nsd/Tizen.Network.Nsd/DnssdService.cs

```diff
@@ -218,14 +220,33 @@ public IPAddressInformation IP
             }
         }
 
-        private void GetTxtRecord(out ushort length, out byte[] value)
+        /// <summary>
+        /// Returns all raw TXT records.
+        /// </summary>
+        /// <remarks>
+        /// After adding the TXT record call for get TXT record.
+        /// </remarks>
+        /// <returns>Returns empty bytes in case of no records available else returns raw text tecords.</returns>
+        /// <since_tizen> 9 </since_tizen>
+        /// <feature>http://tizen.org/feature/network.service_discovery.dnssd</feature>
+        /// <exception cref="NotSupportedException">Thrown when DNS-SD is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when any other error occurred.</exception>
+        public byte[] GetTXTRecords()
```

Comment:
I wonder - why plural "Records"? We are returning one byte array. You have SetRecord and UnsetRecord. RFC 6763 calls it "TXT Record".

Reply:
Native CAPI returns a list of TXT records.
We want the app to parse the TXT records and use them according to TXT record format in DNS-SD specification.
(Section 6.3 in http://files.dns-sd.org/draft-cheshire-dnsext-dns-sd.txt)